src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/ReadAndUpdateOperation.java

```diff
@@ -143,7 +187,7 @@ private Uni<DocumentId> updatedDocument(
               if (result.getRows(0).getValues(0).getBoolean()) {
                 return Uni.createFrom().item(writableShreddedDocument.id());
               } else {
-                return Uni.createFrom().nullItem();
+                throw new LWTException(ErrorCode.CONCURRENCY_FAILURE);
```

Comment:
Not sure if possible, but would be great to indicate doc id of the failing document. Unless maybe it is reported at higher level and need not be included in exception message here?

Reply:
The document id is added along with the exception while creating UpdatedDocument object.